README.md
# Admin Footer Version (rebranded)

[![Build Status](https://travis-ci.org/luciano-croce/admin-footer-version-rebranded.svg?branch=master)](https://travis-ci.org/luciano-croce/admin-footer-version-rebranded)

Available only on https://wordpress.org/plugins/admin-footer-version-rebranded/

Plugin approved in the repository of the plugin directory on 2017-12-25

Development for this plugin takes place at [GitHub](https://github.com/luciano-croce/admin-footer-version-rebranded/) and [Travis CI](https://travis-ci.org/luciano-croce/admin-footer-version-rebranded/).

Compatible with [GlotPress Translations](https://translate.wordpress.org/projects/wp-plugins/admin-footer-version-rebranded). 

Show the **rebranded version** in the admin footer, (dashboard) when it is activated, or automatically, if it is in mu-plugins directory. Expand, secure, enhance, core code on /wp-admin/includes/update.php

**This WordPress plugin expand, secure, enhance, core code on /wp-admin/includes/update.php**

Expand

Necessary

* Same dashboard messages for all WordPress releases.
* All readable text of this plugin are code free.
* Same code work from WP 2.3+ to 4.9+ ~ 5.0-alpha
* Centralized deployments and development for all installations.

Enhance

Includes

* Visual color messages for: development, upgrade, latest.
* Support Single, Multiuser (MU) and Multisite (MS) release.
* Checking PHP and WP requirements before plugin activation.
* No HTML tags have inserted (showing) in Text Domain strings.

Secure

Provides

* Disabled for all users except admins, ability to visualize version.
* Only user roles manage options or core can update WordPress.
* Corrected the missing user role code for core updates.
* Corrected the missing user role code for development updates.
* Secured the user roles that can update WordPress.
* Only users with manage_options capability (WP 2.3+ to 2.9+)
* Only users with update_core capability (WP 3.0+ to 4.9+ ~ 5.0-α)

Table of classifications

WordPress

 * Legacy version 2.3+ to 2.6+ - red color highlighted
 * Oldest version 2.7+ to 2.9+ - red color highlighted
 * Old version 3.0+ to 3.6+ - red color highlighted
 * Supported version 3.7+ to 4.8+ - orange color highlighted
 * Latest version 4.9+ - black color highlighted
 * Development version 5.0-alpha - orange color highlighted

PHP

 * 5.2+ slow - deprecated version red color highlighted
 * 5.3+ slow - deprecated version red color highlighted
 * 5.4+ slow - deprecated version red color highlighted
 * 5.5+ slow - deprecated version red color highlighted
 * 5.6+ slow - acceptable version orange color highlighted
 * 7.0+ fast - supported version black color highlighted
 * 7.1+ fast - supported version black color highlighted
 * 7.2+ fast - supported version black color highlighted
 * 7.3+ fast - development version orange color highlighted

(available on development version)

MySQL

 * 5.0+ oldest version red color highlighted
 * 5.1+ old version red color highlighted
 * 5.5+ acceptable version orange color highlighted
 * 5.6+ current version black color highlighted
 * 5.7+ development version orange color highlighted

(available on development version) 

MariaDB

 * 5.5+ acceptable version orange color highlighted
 * 10.0+ current version black color highlighted

(available on development version)

Tips

A neat trick, is to put this single file admin-footer-version-rebranded.php (not its parent directory) in the /wp-content/mu-plugins/ directory (create it if not exists) so you won't even have to enable it, and will be loaded by default, also, since first step installation of WordPress setup!

Also, for translation functionality, put all files of the single languages (admin-footer-version-rebranded-en_US.mo for example) that you need (not its parent directory) in the /wp-content/mu-plugins/ directory (create it if not exists) and will be loaded by default, they also, since first step installation of WordPress setup!

About it

The code of this plugin, is not written with a PHP framework, but with a simple PHP editor, manually, compatible UTF-8 without BOM, and Unix LF. To avoid code corruption, do not edit it with WordPress embedded editor, or with any specifications incompatible editor.

Translation

Translated by [Luciano Croce](https://profiles.wordpress.org/luciano-croce/) on 14 languages and related formal/informal variants.

Available in English, French, German, Italian, Portuguese, Spanish, and more other languages coming soon.

Acknowledgements

Thanks for the corrections, suggestions, and translation approval to

Team

 * de_CH
 * de_DE
 * en_AU
 * en_CA
 * en_GB
 * en_NZ
 * en_ZA
 * es_ES
 * fr_BE
 * fr_CA
 * fr_FR
 * it_IT
 * pt_BR
 * pt_PT

People

 * [Andrea Gandino](https://profiles.wordpress.org/andg/)
 * [Daniel James](https://profiles.wordpress.org/danieltj/)
 * [Fernando Tellado](https://profiles.wordpress.org/fernandot/)
 * [Felipe Elia](https://profiles.wordpress.org/felipeelia/)
 * [FX Bérnard](https://profiles.wordpress.org/fxbenard/)
 * [Garrett Hyder](https://profiles.wordpress.org/garrett-eclipse/)
 * [Gary Jones](https://profiles.wordpress.org/garyj/)
 * [Laura Sacco](https://profiles.wordpress.org/lasacco/)
 * [Pascal Birchler](https://profiles.wordpress.org/swissspidy/)
 * [Pascal Casier](https://profiles.wordpress.org/casiepa/)
 * [Pedro Mendonça](https://profiles.wordpress.org/pedromendonca/)
 * [Petya Raykovska](https://profiles.wordpress.org/petya/)
 * [Presskopp](https://profiles.wordpress.org/presskopp/)
 * [Stephen Edgar](https://profiles.wordpress.org/netweb/)
 * [Johannes Jülg](https://profiles.wordpress.org/timse201/)
 * [WebAware](https://profiles.wordpress.org/webaware/)
 * [Naoko Takano](https://profiles.wordpress.org/nao/)

Disclaimer

With the exception of the main descriptions, it is considered a good practice, not to translate the readme.txt file in the various languages different from en_US (except for the other english languages derived from english) to standardize the understanding of its technical content, and not to introduce new errors due to the misunderstanding of its correct translation.


CHANGES.md
Future Release Planning

    Adds Settings Options Panel on Admin Menu of Dashboard
    Enable/Disable Footer Text (for users and admins)
    Enable/Disable Footer Text (only for users)
    Enable/Disable Footer Text (only for admins)
    Enable/Disable Footer Update (for users and admins)
    Enable/Disable Footer Update (only for users)
    Enable/Disable Footer Update (only for admins)
    Show default Footer Text (for users and admins)
    Show default Footer Text (only for users)
    Show default Footer Text (only for admins)
    Show default Footer Update (for users and admins)
    Show default Footer Update (only for users)
    Show default Footer Update (only for admins)

1.2.3-dev

Release Date: April 22, 2018

    New Rebranded Admin Footer is now based on WP + PHP + SQL version
    Dismiss unnecessary Serve Happy Dashboard Widget
    Preemptive Support for MySQL 8.0+ or later

1.1.1-dev

Release Date: January 06, 2018

    Adds PHP version recognition
    Adds SQL version recognition
    Adds PHP version on Update Footer
    Adds SQL version on Update Footer

1.0.3

Release Date: April 28, 2018

    Replaced spaces with tabulations to reduce the size of the plugin
    Improved and optimized readme.txt for better understanding
    Updated the translations of language files .mo
    Updated the descriptions to reflect new changes
    Updated the screenshots according new changes
    Updated readme.txt according new changes

1.0.2

Release Date: April 25, 2018

    Revised all texts and descriptions – internationalization
    The descriptions and texts used are more understandable to translate
    Language translations of GlotPress are now more reliable
    Simplify GlotPress translations for polyglots
    Improved the west germanic languages
    Improved the west romance language
    Adds support and description of legacy version concept
    Adds support and description of oldest version concept
    Adds support and description of old version concept
    Adds support and description of supported version concept
    Adds get_locale() plugin meta support – Thx swissspidy

1.0.1

Release Date: December 25, 2017

    Changed plugin name
    Changed plugin description
    Renamed plugin slug
    Renamed text domain slug
    Preemptive Support for WP 5.0-alpha
    Preemptive Support for PHP 7.3+
    Preemptive Support for MySQL 5.7+
    Compatible with WordPress 4.9+
    Work with WordPress 2.3+ to 4.9+ ~ 5.0-alpha
    Make sure that run under WP 2.3+ or greater
    Compatible with PHP 5.2.4+ to 7.2+ ~ 7.3+
    Work with PHP 5.2.4+ to 7.2+ ~ 7.3+
    Make sure that run under PHP 5.2.4+ or greater
    Compatible with MySQL 5.0.15+ to 5.6+ ~ 5.7+ or greater
    Work with MySQL 5.0.15+ to 5.6+ ~ 5.7+ or greater
    Make sure that run under MySQL 5.0.15+ or greater
    Compatible with MariaDB 5.5+ and 10.0+ or greater
    Compatible with GlotPress translations
    Compatible with load_plugin_textdomain Function
    Compatible with load_muplugin_textdomain Function

1.0.0

Release Date: December 12, 2013

    First initial release of the plugin
    Use the filter hook and not the action hook
    Compatible with WordPress 2.3+ or greater
    Requires WordPress Version 2.3+ or greater
    Compatible with PHP 5.2.4+ or greater
    Requires PHP Version 5.2.4+ or greater
    Compatible with MySQL 5.0.15+ or greater
    Requires MySQL 5.0.15+ or greater
    Compatible with WordPress Multisite (MS)
    Compatible with WordPress Multiuser (MU)
    Compatible with mu-plugins directory
    Compatible with 1st step installation setup
    Compatible with the Famous 5 Minute Installation


footer-version-rebranded.php
<?php 
/*
Plugin Name:       Admin Footer Version (rebranded)
Plugin URI:        https://github.com/luciano-croce/admin-footer-version/
Description:       Show rebranded version in admin footer (dashboard backend) when is activated, or automatically, if it is in mu-plugins directory.
Version:           1.0.1
Requires at least: 3.0
Tested up to:      5.0
Requires PHP:      5.2.4
Author:            Luciano Croce
Author URI:        https://github.com/luciano-croce/
License:           GPLv2 or later (license.txt)
License URI:       https://www.gnu.org/licenses/gpl-2.0.html
Text Domain:       admin-footer-version
Domain Path:       /languages
Network:           true
GitHub Plugin URI: https://github.com/luciano-croce/admin-footer-version/
GitHub Branch:     master
Requires WP:       3.0
 *
 * Copyright 2013-2017 Luciano Croce (email: luciano.croce@gmail.com)
 *
 * This program is free software; you can redistribute it and/or modify it under the terms of the GNU
 * General Public License version 2, as published by the Free Software Foundation. You may NOT assume
 * that you can use any other version of the GPL.
 *
 * This program is distributed in the hope that it will be useful, on an "AS IS", but WITHOUT ANY WARRANTY;
 * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 *
 * This program is written with the intent of being helpful,
 * but you are responsible for its use or actions on your own website.
 *
 * Tips - A neat trick, is to put this single file admin-footer-version.php (not its parent directory)
 * in the /wp-content/mu-plugins/ directory (create it if not exists) so you won't even have to enable it,
 * and will be loaded by default, also, since first step installation of WordPress setup!
 *
 * The code of this plugin is not written with a php framework, but with a simple php editor, manually.
 */

	/**
	 * Admin Footer Version (rebranded)
	 *
	 * Show rebranded version in admin footer (dashboard backend)
	 *
	 * PHPDocumentor
	 *
	 * @package    WordPress\Plugin
	 * @subpackage Footer\Admin_Footer_Version
	 * @link       https://github.com/luciano-croce/admin-footer-version/
	 * @version    1.0.1 (Build 2017-12-25) Stable
	 * @since      1.0.0 (Build 2013-12-12) 1st Release
	 * @author     Luciano Croce <luciano.croce@gmail.com>
	 * @copyright  2013-2017 - Luciano Croce
	 * @license    https://www.gnu.org/licenses/gpl-2.0.html - GPLv2 or later
	 * @todo       Preemptive support for WordPress 5.0-alpha
	 */

/**
 * Prevent direct access to plugin files.
 *
 * For security reasons, exit without any notifications:
 * - without show the details of the system
 * - without warn the existence of this plugin
 * - show the generic header 403 forbidden error
 * - close the connection header
 *
 * @author  Luciano Croce <luciano.croce@gmail.com>
 * @version 1.0.1 (Build 2017-12-25)
 * @since   1.0.0 (Build 2013-12-12)
 */
if ( ! defined( 'ABSPATH' ) ) exit;
if ( ! defined(  'WPINC'  ) ) exit;

if ( ! function_exists( 'add_action' ) ) {
	header( 'HTTP/0.9 403 Forbidden' );
	header( 'HTTP/1.0 403 Forbidden' );
	header( 'HTTP/1.1 403 Forbidden' );
	header( 'HTTP/1.2 403 Forbidden' );
	header( 'HTTP/1.3 403 Forbidden' );
	header( 'Status: 403 Forbidden'  );
	header( 'Connection: Close'      );
		exit;
}

add_filter( 'update_footer', 'core_update_footer_rebranded', 11 ); # By default core update_footer has level 10

/**
 * Expand, secure, enhance, core code, on /wp-admin/includes/update.php
 *
 * @author  Luciano Croce <luciano.croce@gmail.com>
 * @version 1.0.1 (Build 2017-12-25)
 * @since   WordPress 3.0+ (single and multisite)
 *
 * @param   string $msg
 * @return  string
 */
function core_update_footer_rebranded( $msg = '' ) {

	if ( ! current_user_can( 'update_core' ) ) {
//		return sprintf( __( 'Version %s' ), get_bloginfo( 'version', 'display' ) ); # Original core code on /wp-admin/includes/update.php
		return sprintf( __( 'You do <u>not have access</u> to this information <strong>for security purpose</strong>.', 'admin-footer-version' ), get_bloginfo( 'version', 'display' ) ); # Sorry, only users can update core have access to this information.
	}

	$cur = get_preferred_from_update_core();
	if ( ! is_object( $cur ) ) {
		$cur = new stdClass;
	}

	if ( ! isset( $cur->current ) ) {
		$cur->current = '';
	}

	if ( ! isset( $cur->url ) ) {
		$cur->url = '';
	}

	if ( ! isset( $cur->response ) ) {
		$cur->response = '';
	}

	switch ( $cur->response ) {
		case 'development':
			/* Translators - 1: WordPress version number 2: WordPress updates admin screen URL */
//			return sprintf( __( 'You are using a development version (%1$s). Cool! Please <a href="%2$s">stay updated</a>.' ), get_bloginfo( 'version', 'display' ), network_admin_url( 'update-core.php' ) ); # Original core code on /wp-admin/includes/update.php
			return sprintf( __( 'You are using a <u>Development Version</u> %1$s', 'admin-footer-version' ) . ' - ' . __( '<strong><a href="%2$s">Stay Updated</a></strong>', 'admin-footer-version' ), get_bloginfo( 'version', 'display' ), network_admin_url( 'update-core.php' ) );

		case 'upgrade':
//			return '<strong><a href="' . network_admin_url( 'update-core.php' ) . '">' . sprintf( __( 'Get Version %s' ), $cur->current ) . '</a></strong>'; # Original core code on /wp-admin/includes/update.php
			return sprintf( __( 'You are using a <u>not up to date</u> Version %1$s', 'admin-footer-version' ), get_bloginfo( 'version', 'display' ), network_admin_url( 'update-core.php' ) ) . ' - ' . '<strong><a href="' . network_admin_url( 'update-core.php' ) . '">' . sprintf( __( 'Get Version %s', 'admin-footer-version' ), $cur->current ) . '</a></strong>';

		case 'latest':
		default:
//			return sprintf( __( 'Version %s' ), get_bloginfo( 'version', 'display' ) ); # Original core code on /wp-admin/includes/update.php
			return sprintf( __( 'You are using a <u>latest available</u>', 'admin-footer-version' ) . ' ' . __( 'Version %s', 'admin-footer-version' ), get_bloginfo( 'version', 'display' ) );
	}

}
